app/actions/pricesActions.js

```diff
@@ -2,32 +2,43 @@
 import axios from 'axios'
 import { createActions } from 'spunky'
 
-import { DEFAULT_CURRENCY_CODE } from '../core/constants'
+import { getDefaultTokens } from '../core/nep5'
+import { DEFAULT_CURRENCY_CODE, ASSETS } from '../core/constants'
 
 type Props = {
   currency?: string
 }
 
-function mapPrices(tickers, currency) {
+function mapPrices(pricingData: Array<any>, currency) {
   const mapping = {}
-
-  tickers.forEach(ticker => {
-    mapping[ticker.symbol] = parseFloat(
-      ticker[`price_${currency.toLowerCase()}`]
-    )
+  const upperCasedCurrency = currency.toUpperCase()
```

Comment:
this method should just be a reduce in my opinion rather than forEach building a new map

Reply:
updated to use reduce thanks